libmscore/element.cpp

```diff
@@ -2149,25 +2178,155 @@ void Element::autoplaceSegmentElement(qreal minDistance)
       }
 #endif
 
+//---------------------------------------------------------
+//   setOffsetChanged
+//---------------------------------------------------------
+
+void Element::setOffsetChanged(bool v, bool absolute, const QPointF& diff)
+      {
+      if (v)
+            _offsetChanged = absolute ? 1 : -1;
+      else
+            _offsetChanged = 0;
+      _changedPos = pos() + diff;
+      }
+
+//---------------------------------------------------------
+//   rebaseOffset
+//    calculates new offset for moved elements
+//    for drag & other actions that result in absolute position, apply the new offset
+//    for nudge & other actions that result in relative adjustment, return the vertical difference
+//---------------------------------------------------------
+
+qreal Element::rebaseOffset(bool nox)
+      {
+      QPointF off = offset();
+      QPointF p = _changedPos - pos();
+      if (nox)
+            p.rx() = 0.0;
+      bool saveChangedValue = _offsetChanged;
+
+      if (staff() && propertyFlags(Pid::PLACEMENT) != PropertyFlags::NOSTYLE) {
+            // check if flipped
+            // TODO: elements that support PLACEMENT but not as a styled property
+            QRectF r = bbox().translated(_changedPos);
+            qreal staffHeight = staff()->height();
+            Element* e = isSpannerSegment() ? toSpannerSegment(this)->spanner() : this;
+            bool multi = e->isSpanner() && toSpanner(e)->spannerSegments().size() > 1;
+            bool above = e->placeAbove();
+            bool flipped = above ? r.top() > staffHeight : r.bottom() < 0.0;
+            if (flipped && !multi) {
+                  off.ry() += above ? -staffHeight : staffHeight;
+                  undoChangeProperty(Pid::OFFSET, off + p);
+                  _offsetChanged = saveChangedValue;
```

Comment:
Ouch, definitely a mistake!  Will fix.  I'm sort of thinking it is actually *correct* for it to always restore positive, since offset adjustment performed here may be more consistent with that.  Need to think that through more.